### DISCUSSION TITLE
`enso-types` is sometimes rebuilt superflously

### DISCUSSION THREAD

**Discussion by farmaazon** [enso-org/enso#5681] [Work In Progress - GUI] [answered]
Sometimes when I switch between building in console and "clipping" in the IDE, I see that `enso-types` crate is rebuilt many times.

And that is caused by `build.rs` script which regenerates file _in src directory_. When I switch to another configuration (or another target dir, as in IntelliJ configs), then that another configuration thinks that something has changed in our source code.

So the question is: why `enso-types` does not generate source code in $OUT dir (according to the recommendations)? The dim_macros module could just have proper include inside.

Calling @wdanilo - the author of the script

(BTW, we also put this generated file to git repository)

**Comment by wdanilo** [comment, marked as answer]
I don't know TBH, but seems like a stupid mistake indeed. We should refactor it and make it better.

**Reply by farmaazon** [reply]
Thanks for reply. I made a task for doing this https://github.com/enso-org/enso/issues/5686 I will do it "in the meantime" next week. Or just I'll see the reason it's been done this way.

**Reply by wdanilo** [reply]
Btw, thanks for discovering it. I've seen this behavior but I did not dig deep enough to understand why it happens.
